sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -66,7 +66,15 @@
 ParamsType = Mapping[str, Union[PrimitiveData, Sequence[PrimitiveData]]]
 
 FileContent = Union[str, bytes, IO[str], IO[bytes]]
-FileType = Tuple[Optional[str], FileContent]
+
+FileType = Union[
+    # file (or bytes)
+    FileContent,
+    # (filename, file (or bytes))
+    Tuple[Optional[str], FileContent],
+    # (filename, file (or bytes), content_type)
+    Tuple[Optional[str], FileContent, Optional[str]],
+]
 
 FilesType = Union[Mapping[str, FileType], Sequence[Tuple[str, FileType]]]
```

Comment:
filename is not a required input to `FileType`, since you can just pass in the file directly. `FilesType` is the type we pass to the `files` keyword arg in `azure.core.rest.HttpRequest`, where you pass some sort of mapping / sequence for a field to a `FileType` value, i.e. here https://requests.readthedocs.io/en/latest/user/quickstart/#post-a-multipart-encoded-file